distribution/openrct2.d.ts

```diff
@@ -720,20 +720,25 @@ declare global {
      * Represents a guest or staff member.
      */
     interface Peep extends Entity {
+        /**
+         * Whether the peep is a guest or staff member.
+         */
+        peepType: PeepType;
+
         /**
          * Name of the peep.
          */
         name: string;
 
         /**
-         * Colour of the peep's t-shirt.
+         * Main flags for peep behaviour.
          */
-        tshirtColour: number;
+        flags: number;
```

Comment:
While there are advantages to opening the flags up like this, there are a few drawbacks as well. Knowledge of which bit is meant for what flag is one. However, a more pressing one is that any plug-ins using these flags will break if we ever decide to change or add flags, right? The main code base will be refactored accordingly, but we won't have that kind of control over plug-ins.

Reply:
What do you suggest doing instead? It takes so long going through every field, and any large enumeration like this then requires expanding to many more fields all which require code and documentation or writing a routine to convert to and from strings.

Reply:
What about this way? (see latest commit)

Reply:
Your latest commit has introduced a much better approach, I think. These two functions should be much more extensible and easier to support in the future, I think. Thanks. :)